File: DaggerProject3/app/src/main/java/com/example/daggerproject3/module/CameraModule.java
package com.example.daggerproject3.module;

import android.util.Log;

import com.example.daggerproject3.model.Camera;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;

@Module
public abstract class CameraModule {

    int megaPixel;

    public CameraModule(int megaPixel) {
        this.megaPixel = megaPixel;
    }

    @Provides
    static Camera getCamera(@Named("megaPixel") int megaPixel){
        return new Camera(megaPixel);
    }
}

File: DaggerProject3/app/src/main/java/com/example/daggerproject3kotlin/module/CameraModule.kt
package com.example.daggerproject3kotlin.module

import com.example.daggerproject3kotlin.model.Camera
import dagger.Module
import dagger.Provides
import javax.inject.Named

@Module
class CameraModule {

    @Provides
    fun getCamera(@Named("megaPx") megaPx:Int):Camera{
        return Camera(megaPx)
    }
}